dome9/resource_dome9_image_assurance_policy.go

```diff
@@ -0,0 +1,129 @@
+package dome9
+
+import (
+	"log"
+
+	"github.com/dome9/dome9-sdk-go/dome9/client"
+	"github.com/dome9/dome9-sdk-go/services/imageassurance/imageassurance_policy"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+)
+
+func resourceImageAssurancePolicy() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceImageAssurancePolicyCreate,
+		Read:   resourceImageAssurancePolicyRead,
+		Update: resourceImageAssurancePolicyUpdate,
+		Delete: resourceImageAssurancePolicyDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Schema: map[string]*schema.Schema{
+			"target_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"target_type": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringInSlice([]string{"Environment", "OrganizationalUnit"}, false),
+			},
+			"ruleset_id": {
+				Type:     schema.TypeInt,
+				Required: true,
+			},
+			"notification_ids": {
+				Type:     schema.TypeList,
+				Required: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+			"admission_control_action": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validation.StringInSlice([]string{"Prevention", "Detection"}, false),
+			},
+			"admission_control_unscanned_action": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validation.StringInSlice([]string{"Prevention", "Detection"}, false),
+			},
+		},
+	}
+}
+
+func resourceImageAssurancePolicyCreate(d *schema.ResourceData, meta interface{}) error {
+	d9Client := meta.(*Client)
+	req := expandImageAssurancePolicyRequest(d)
+	log.Printf("[INFO] Creating ImageAssurance policy request %+v\n", req)
+	resp, _, err := d9Client.imageAssurancePolicy.Create(&req)
+	if err != nil {
+		return err
+	}
+
+	log.Printf("[INFO] Created ImageAssurance Policy with ID: %v\n", resp.ID)
+	d.SetId(resp.ID)
+
+	return resourceImageAssurancePolicyRead(d, meta)
+}
+
+func expandImageAssurancePolicyRequest(d *schema.ResourceData) imageassurance_policy.ImageAssurancePolicyRequest {
+	return imageassurance_policy.ImageAssurancePolicyRequest{
+		TargetId:                        d.Get("target_id").(string),
+		RulesetId:                       d.Get("ruleset_id").(int),
+		NotificationIds:                 expandNotificationIDs(d, "notification_ids"),
+		TargetType:                      d.Get("target_type").(string),
+		AdmissionControllerAction:       d.Get("admission_control_action").(string),
```

Comment:
cast on potential nil?

Reply:
in AC the action is required.
what ` d.Get("admission_control_action")` returns? if it can be nil then `nil.(string)` will cause panic